### ISSUE TITLE
Decoding JSON is Corrupted

### ISSUE THREAD

**Issue by snagglegrolop** [SwiftcordApp/DiscordKit#37]
```terminal
2023-11-30T21:57:49-0800 info RobustWebSocket : version=10 ws=wss://gateway.discord.gg/?v=10&encoding=json&compress=zlib-stream [DiscordKitCore] [CONNECT]
2023-11-30T21:57:49-0800 debug RobustWebSocket : reason=connection is reachable [DiscordKitCore] Reset backoff
2023-11-30T21:57:49-0800 warning RobustWebSocket : [DiscordKitCore] Reconnection in progress, not reconnecting
2023-11-30T21:57:49-0800 info RobustWebSocket : [DiscordKitCore] [CONNECTED]
2023-11-30T21:57:49-0800 trace DecompressionEngine : decompressed.count=109 original.count=114 [DiscordKitCore] Decompressed data
2023-11-30T21:57:49-0800 warning RobustWebSocket : error=dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid JSON.", underlyingError: Optional(Error Domain=NSCocoaErrorDomain Code=3840 "Unexpected character 'ï' around line 2, column 1." UserInfo={NSJSONSerializationErrorIndex=1, NSDebugDescription=Unexpected character 'ï' around line 2, column 1.}))) [DiscordKitCore] Error decoding message
2023-11-30T21:57:53-0800 warning RobustWebSocket : timeout=4.0 [DiscordKitCore] Connection timed out
2023-11-30T21:57:53-0800 warning RobustWebSocket : [DiscordKitCore] Forcibly closing connection
2023-11-30T21:57:53-0800 info RobustWebSocket : [DiscordKitCore] [RECONNECT] Preemptively attempting reconnection
2023-11-30T21:57:53-0800 info RobustWebSocket : attempt=1 connectIn=3.8000000000000003 [DiscordKitCore] Retrying connection
2023-11-30T21:57:53-0800 warning RobustWebSocket : [DiscordKitCore] Already reconnecting, not reconnecting
2023-11-30T21:57:53-0800 warning RobustWebSocket : [DiscordKitCore] [WS CLOSED] closeCode: 1006
2023-11-30T21:57:53-0800 error RobustWebSocket : [DiscordKitCore] [WS CLOSED] closed with error: no error description
2023-11-30T21:57:53-0800 warning RobustWebSocket : [DiscordKitCore] Already reconnecting, not reconnecting
2023-11-30T21:57:53-0800 error RobustWebSocket : error=The operation couldn’t be completed. Socket is not connected [DiscordKitCore] Receive error
2023-11-30T21:57:53-0800 warning RobustWebSocket : [DiscordKitCore] Forcibly closing connection```

**Comment by Kiszaner** [comment]
I was having the same issue using bot-support branch, but it seems to be already fixed on main

**Comment by cryptoAlgorithm** [comment]
That's right, this issue has been fixed with commit 2d42c69cafe592300a1a9d3a307bf485294026c7 on `main`. It's not recommended to use `bot-support` as it's now a stale branch created for initial efforts to bring bot support to DiscordKit. The documentation on GitBook has been updated a while back to reflect this too.
